boards/components/src/gpio.rs

```diff
@@ -6,19 +6,20 @@
 //! let gpio = components::gpio::GpioComponent::new(
 //!     board_kernel,
 //!     components::gpio_component_helper!(
+//!         // pin struct type
 //!         sam4l::gpio::GPIOPin,
-//!         &nrf52840::gpio::PORT[GPIO_D2],
-//!         &nrf52840::gpio::PORT[GPIO_D3],
-//!         &nrf52840::gpio::PORT[GPIO_D4],
-//!         &nrf52840::gpio::PORT[GPIO_D5],
-//!         &nrf52840::gpio::PORT[GPIO_D6],
-//!         &nrf52840::gpio::PORT[GPIO_D7],
-//!         &nrf52840::gpio::PORT[GPIO_D8],
-//!         &nrf52840::gpio::PORT[GPIO_D9],
-//!         &nrf52840::gpio::PORT[GPIO_D10]
+//!         // pin number => pin reference
+//!         2 => &nrf52840::gpio::PORT[GPIO_D2],
+//!         3 => &nrf52840::gpio::PORT[GPIO_D3],
+//!         4 => &nrf52840::gpio::PORT[GPIO_D4],
+//!         5 => &nrf52840::gpio::PORT[GPIO_D5],
+//!         6 => &nrf52840::gpio::PORT[GPIO_D6],
+//!         7 => &nrf52840::gpio::PORT[GPIO_D7],
+//!         8 => &nrf52840::gpio::PORT[GPIO_D8],
+//!         9 => &nrf52840::gpio::PORT[GPIO_D9],
+//!         10 => &nrf52840::gpio::PORT[GPIO_D10]
```

Comment:
I think the gpio_component_helper macro needs documentation describing that numbers skipped are made `None` to allow for continuity of pin mapping.

Also, this example usage doesn't seem to match the initialization actually used by the nrf52840.

Reply:
> I think the gpio_component_helper macro needs documentation describing that numbers skipped are made `None` to allow for continuity of pin mapping.
> 
Added it.

> Also, this example usage doesn't seem to match the initialization actually used by the nrf52840.

That was the original example, most probably the boards have changed since then. I updated it with a new example.